x-pack/platform/plugins/shared/rule_registry/server/routes/bulk_update_tags.ts

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { IRouter } from '@kbn/core/server';
+
+import { schema } from '@kbn/config-schema';
+import { transformError } from '@kbn/securitysolution-es-utils';
+import type { RacRequestHandlerContext } from '../types';
+import { BASE_RAC_ALERTS_API_PATH } from '../../common/constants';
+import {
+  MAX_ALERT_IDS_PER_REQUEST,
+  MAX_INDEX_NAME,
+  MAX_QUERY_LENGTH,
+  MAX_TAGS_TO_UPDATE,
+} from '../alert_data_client/constants';
+
+const bodySchema = schema.object({
+  index: schema.string({ maxLength: MAX_INDEX_NAME }),
+  add: schema.maybe(schema.arrayOf(schema.string(), { minSize: 1, maxSize: MAX_TAGS_TO_UPDATE })),
+  remove: schema.maybe(
+    schema.arrayOf(schema.string(), { minSize: 1, maxSize: MAX_TAGS_TO_UPDATE })
+  ),
+  alertIds: schema.maybe(
+    schema.arrayOf(schema.string(), { minSize: 1, maxSize: MAX_ALERT_IDS_PER_REQUEST })
+  ),
+  query: schema.maybe(schema.string({ maxLength: MAX_QUERY_LENGTH })),
+});
+
+export const bulkUpdateTagsRoute = (router: IRouter<RacRequestHandlerContext>) => {
+  router.post(
```

Comment:
@baileycash-elastic is the method correct? Asking only because in the PR description you mentioned PATCH